### PR TITLE
✅ [RUM] introduce test setup builder

### PR DESCRIPTION
**Pull request by bcaudan**
## Motivation

Ease bootstrapping of RUM unit tests by:

- avoid to wire test subjects with all their dependencies
- abstract mock setup (fake server, clock, ...) 
- mutualize cleanup

```ts
beforeEach(() => {
  setupBuilder = setup()
    .withFakeServer()
    .withRum()
    .withViewCollection()
})

afterEach(() => {
  setupBuilder.cleanup()
})

it('should ...', () => {
  const { server, lifeCycle } = setupBuilder.build()
  server.requests = []

  lifeCycle.notify(...)

  ...
})


```

## Changes

- remove a useless test
- introduce test setup builder
- use it in existing tests when relevant

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/master/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum/test/viewCollection.spec.ts [hunk 1]
```diff
@@ -116,23 +119,32 @@ describe('rum track renew session', () => {
...
+  let addRumEvent: jasmine.Spy
   let getViewEvent: (index: number) => View
 
   beforeEach(() => {
-    jasmine.clock().install()
-    ;({ lifeCycle, getViewEvent } = spyOnViews())
+    ;({ addRumEvent, getViewEvent } = spyOnViews())
+
+    const fakeLocation: Partial<Location> = { pathname: '/foo' }
+    mockHistory(fakeLocation)
```
Comment (BenoitZugmeyer) [inline packages/rum/test/viewCollection.spec.ts:130]: Maybe `mockHistory` could be part of the setup builder?

File: packages/rum/test/rum.spec.ts [hunk 2]
```diff
@@ -452,36 +440,34 @@ describe('rum global context', () => {
   })
 
   it('should not be automatically snake cased', () => {
-    RUM.setRumGlobalContext({ fooBar: 'foo' })
+    const { server, lifeCycle, rumApi } = setupBuilder.build()
+    server.requests = []
```
Comment (BenoitZugmeyer) [inline packages/rum/test/rum.spec.ts:444]: I've seen this multiple times.. The server can receive requests between its creation and the test execution? I'd be in favor of reseting the server requests in a setup builder "build task".

File: packages/rum/test/viewCollection.spec.ts (hunk 1 above)
Comment (bcaudan) [reply packages/rum/test/viewCollection.spec.ts:130]: I think we can still have helper specific to spec files when it is not used elsewhere.
I'd be in favor of moving it there if we need to reuse it in another context.

File: packages/rum/test/rum.spec.ts (hunk 2 above)
Comment (bcaudan) [reply packages/rum/test/rum.spec.ts:444]: it depends if you are interested or not in the events generated when starting rum.

File: packages/rum/test/viewCollection.spec.ts (hunk 1 above)
Comment (BenoitZugmeyer) [reply packages/rum/test/viewCollection.spec.ts:130]: Ok!

File: packages/rum/test/rum.spec.ts (hunk 2 above)
Comment (BenoitZugmeyer) [reply packages/rum/test/rum.spec.ts:444]: Ah, I see! All right.

**Pull request merged**